tsdb/tombstones/tombstones.go

```diff
@@ -252,6 +252,26 @@ func (t *MemTombstones) Get(ref uint64) (Intervals, error) {
 	return t.intvlGroups[ref], nil
 }
 
+func (t *MemTombstones) DeleteTombstones(ref uint64) {
```

Comment:
Taking the slice of ref will avoid frequent locking and unlocking (similar to how we delete postings).

```suggestion
func (t *MemTombstones) DeleteTombstones(ref []uint64) {
```